### ISSUE TITLE
problem with warp if the `x` and `flo` are different sizes

### ISSUE THREAD

**Issue by scott-vsi** [PruneTruong/DenseMatching#28]
I think there is a problem with `utils_flow/pixel_wise_mapping.py:warp` [[link](https://github.com/PruneTruong/DenseMatching/blob/b89ec4b446816f596e36302c150aa3b732266c36/utils_flow/pixel_wise_mapping.py#L58)] if `x`, the image to warp, im2, and `flo`, the dense flow, are different sizes. 

`flo` has shape [H_1, W_1, xy] with x-range (-W_2, W_2) and y-range (-H_2, H_2) and, as noted in the documentation, maps `im2` (aka `x`) with shape [H_2, W_2] back to `im1` with shape [H_1, W_1] (via torch.nn.functional.grid_sample). However, here `flo` is rescaled to [-1, 1] by H_1, W_1, not H_2, W_2 as it should be.
Perhaps this limitation is suggested by the documentation, since it says the shape of x and flo are both (H,W), but that should be made explicit and even ensured with an assert I think.

This affects several other functions: `utils_flow/pixel_wise_mapping.py:warp_with_mapping`, and `unormalise_and_convert_mapping_to_flow`, `unormalise_flow_or_mapping`, `unnormalize`, & `normalize` in `utils_flow/flow_and_mapping_operations.py`. `utils_flow/pixel_wise_mapping.py:remap_using_flow_fields` works as expected because it does not rescale the flow.

**Comment by PruneTruong** [comment]
I always only used these functions when both the images and the flow are the same size. But you're right that it could be risky to have it like this for other cases. I will update later, thanks.
